src/System.Net.Http.Formatting/HttpMessageContent.cs

```diff
@@ -354,7 +357,7 @@ private byte[] SerializeHeader()
             return Encoding.UTF8.GetBytes(message.ToString());
         }
 
-        private void ValidateStreamForReading(Stream stream)
+        private void ValidateStreamForReading(Stream stream) // ???
```

Comment:
Yeah, this seems like a bug, it should have checked CanSeek before setting the Position, not CanRead. I have seen some Streams return CanRead false after they're done, so that might have avoided the bug.